src/Authentication.Abstractions/AzureSession.cs

```diff
@@ -234,16 +238,16 @@ public void RegisterComponent<T>(string componentName, Func<T> componentInitiali
                 () =>
                 {
                     var key = new ComponentKey(componentName, typeof(T));
-                    if (!_componentRegistry.ContainsKey(key) || overwrite)
+                    if (!_componentRegistry.ContainsKey(key) || overwrite) // only proceed if key not found or overwrite is true
                     {
-                        if (_componentRegistry.ContainsKey(key) && overwrite)
+
+                        if (overwrite
+                            && _componentRegistry.TryGetValue(key, out var existed)
+                            && existed is IAzureSessionListener existedListener)
                         {
-                            var existed = _componentRegistry[key];
-                            if (existed is IAzureSessionListener existedListener)
-                            {
-                                _eventHandler -= existedListener.OnEvent;
-                            }
+                            _eventHandler -= existedListener.OnEvent;
                         }
+
                         var component = componentInitializer();
                         _componentRegistry[key] = component;
```

Comment:
The scenario here matches [Store a key/value pair in the dictionary unconditionally, and overwrite the value of a key that already exists](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-8.0#remarks:~:text=Store%20a%20key/value%20pair%20in%20the%20dictionary%20unconditionally%2C%20and%20overwrite%20the%20value%20of%20a%20key%20that%20already%20exists) so I followed the recommendation to use the setter (`[]`). And since it's unconditional and always overwrites, I figure there's no need to handle the case that it's unsuccessful. So I always remove the handler.